### PR TITLE
`build/pkgs/sympy`: Upgrade to 1.13.2

### PR DESCRIPTION
**Pull request by oscarbenjamin**
See https://groups.google.com/g/sage-devel/c/fOTkNoU02Oo/m/j116LUouBwAJ

This upgrade includes doctest changes needed for compatibility with SymPy 1.13.

This PR is also used by SymPy CI to keep track of compatibility between SymPy and Sage:
https://github.com/sympy/sympy/blob/master/.github/workflows/ci-sage.yml
https://github.com/sympy/sympy/actions/runs/6724078875/job/18275557828

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
This shows the expected test failures:
```
----------------------------------------------------------------------
sage -t --random-seed=... src/sage/doctest/forker.py  # 1 doctest failed
sage -t --random-seed=... src/sage/functions/hypergeometric.py  # 1 doctest failed
sage -t --random-seed=... src/sage/symbolic/expression.pyx  # 1 doctest failed
sage -t --random-seed=... src/sage/typeset/ascii_art.py  # 1 doctest failed
----------------------------------------------------------------------
```
These tests are expected to fail with SymPy 1.12 but should pass with SymPy's master branch.

**Comment by oscarbenjamin** [comment]
I have made SymPy's Sage CI job point to this PR: https://github.com/sympy/sympy/pull/25867

The SymPy CI job has passed:
https://github.com/sympy/sympy/actions/runs/6752438216/job/18357799394

Anyone wanting to check this in future can look at the status of CI on the SymPy master branch:
https://github.com/sympy/sympy/commits/master

image

**Comment by mkoeppe** [comment]
Our PDF docbuild does not like the printing change using "↪" (which you mentioned in https://groups.google.com/g/sage-devel/c/fOTkNoU02Oo/m/j116LUouBwAJ)

**Comment by mkoeppe** [comment]
Commits cfc1728 and 226c15f make the CI use sympy from its master branch

**Comment by oscarbenjamin** [comment]
The changes here seem to have caused the SymPy CI job to fail because the sage build now expects git to be installed:
https://github.com/sympy/sympy/actions/runs/6849261888
```
sage-logger -p 'sage --pip install -r "/sage/build/pkgs/sympy/requirements.txt"' '/sage/logs/pkgs/sympy.log'
[sympy] installing. Log file: /sage/logs/pkgs/sympy.log
[sagelib-10.2.rc1] installing. Log file: /sage/logs/pkgs/sagelib-10.2.rc1.log
  [sympy] error installing, exit status 1. End of log file:
  [sympy]   Collecting git+https://github.com/sympy/sympy#egg_info=sympy (from -r /sage/build/pkgs/sympy/requirements.txt (line 1))
  [sympy]     Cloning https://github.com/sympy/sympy to /tmp/pip-req-build-sjb0fjju
  [sympy]     ERROR: Error [Errno 2] No such file or directory: 'git' while executing command git version
  [sympy]   ERROR: Cannot find command 'git' - do you have 'git' installed and in your PATH?
  [sympy]   Collecting git+https://github.com/sympy/sympy#egg_info=sympy (from -r /sage/build/pkgs/sympy/requirements.txt (line 1))
  [sympy]     Cloning https://github.com/sympy/sympy to /tmp/pip-req-build-0ydw9q2x
  [sympy]     ERROR: Error [Errno 2] No such file or directory: 'git' while executing command git version
  [sympy]   ERROR: Cannot find command 'git' - do you have 'git' installed and in your PATH?
  [sympy] Full log file: /sage/logs/pkgs/sympy.log
make[5]: *** [Makefile:3249: sympy-no-deps] Error 1
make[4]: *** [Makefile:3249: sympy] Error 2
  [pplpy_doc-0.8.9] successfully installed.
  [sagelib-10.2.rc1] successfully installed.
make[4]: Target 'all-sage-docs' not remade because of errors.
make[4]: Target 'all-sage' not remade because of errors.
make[3]: *** [Makefile:2833: all-start] Error 2
make[3]: Leaving directory '/sage/build/make'

real	0m13.882s
user	0m12.792s
sys	0m4.567s
***************************************************************
Error building Sage.
```
I am a bit confused about exactly which version of sympy would be used in the sympy CI job if the build is attempting to install sympy from the master branch (rather than the sympy commit that triggers the CI job).

**Comment by mkoeppe** [comment]
@oscarbenjamin Thanks for reporting this. I've pushed a fix

**Comment by oscarbenjamin** [comment]
The git problem still seems to be happening:
https://github.com/sympy/sympy/actions/runs/7015507471/job/19084955715

File: build/pkgs/sympy/package-version.txt [hunk 1]
```diff
@@ -1 +1 @@
-1.12.1
+1.13.0
```
Comment (oscarbenjamin) [inline build/pkgs/sympy/package-version.txt]: I am assuming that this is what it should be set to now...

There is likely to be a 1.13.1 release coming very soon.

**Comment by tornaria** [comment]
Works fine for me.

**Comment by mkoeppe** [comment]
It cannot be merged in this form (as it changes the package type for testing purposes); it needs to wait for the actual release.

**Comment by mkoeppe** [comment]
As SymPy 1.13.0 is out (congrats @oscarbenjamin), I am reusing this PR for the upgrade

**Comment by tornaria** [comment]
> As SymPy 1.13.0 is out (congrats @oscarbenjamin), I am reusing this PR for the upgrade

Yes, that's why I gave positive review. But it's much better if you can also review since I only tested sagelib.

**Comment by mkoeppe** [comment]
conda tests fail because sympy is pinned to `1.12.1` via conda-lock in the CI.

**Comment by oscarbenjamin** [comment]
I've just pushed SymPy 1.13.1 to PyPI:

https://pypi.org/project/sympy/

I am not sure when it will appear in conda forge (I don't manage that):

https://github.com/conda-forge/sympy-feedstock

**Comment by oscarbenjamin** [comment]
> I've just pushed SymPy 1.13.1 to PyPI:

I've now pushed 1.13.2 to PyPI.

The conda package update is ready but someone needs to merge it:
https://github.com/conda-forge/sympy-feedstock/pull/59

**Comment by mkoeppe** [comment]
I've backed out the attempted update of our conda-lock files from the branch -- other breakage stops the CI Conda from working anyway

**Comment by oscarbenjamin** [comment]
> other breakage stops the CI Conda from working anyway

Is this a Sage CI problem or do you mean that there is an issue with the SymPy Conda package?

**Comment by mkoeppe** [comment]
unrelated to SymPy

**Comment by oscarbenjamin** [comment]
Thanks everyone!

**Comment by tobiasdiez** [comment]
With this PR the conda ci fails now:
```
sage -t --random-seed=338312113839497447506128839798707042314 src/sage/doctest/forker.py  # 1 doctest failed
sage -t --random-seed=338312113839497447506128839798707042314 src/sage/functions/hypergeometric.py  # 1 doctest failed
sage -t --random-seed=338312113839497447506128839798707042314 src/sage/typeset/ascii_art.py  # 1 doctest failed
```

(seen also in the ci of this PR) 
@mkoeppe next time maybe don't set a PR to positive review if you noticed a breaking CI.

Could you please provide a follow-up PR (e.g. by accepting the results of both versions).

**Pull request merged**